src/MCPServer/lib/RPCServer.py

```diff
@@ -54,206 +65,374 @@ class NotFoundError(RPCServerError):
+            payload.get("metadata_set_id"),
+        )
+        kwargs = {
+            "auto_approve": payload.get("auto_approve"),
+            "wait_until_complete": payload.get("wait_until_complete"),
+        }
+        processing_config = payload.get("processing_config")
+        if processing_config is not None:
+            kwargs["processing_config"] = processing_config
+        return create_package(*args, **kwargs).pk
+
+    def _approve_transfer_by_path_handler(self, worker, job, payload):
+        """Approve a transfer matched by its path.
+
+        [config]
+        name = approveTransferByPath
+        raise_exc = no
+        """
+        db_transfer_path = payload.get("db_transfer_path")
+        transfer_type = payload.get("transfer_type", "standard")
+        user_id = payload.get("uid")
+        job = Job.objects.filter(
+            directory=db_transfer_path,
+            currentstep=Job.STATUS_AWAITING_DECISION
+        ).first()
+        if not job:
+            raise NotFoundError("There is no job awaiting a decision.")
+        chain_id = get_approve_transfer_chain_id(transfer_type)
+        try:
+            choicesAvailableForUnits[job.pk].proceedWithChoice(
+                chain_id, user_id)
+        except IndexError:
+            raise NotFoundError("Could not find choice for unit")
+        return job.sipuuid
+
+    def _approve_partial_reingest_handler(self, worker, job, payload):
+        """Approve a partial reingest.
+
+        TODO: (from the old api/views.py function) this is just a temporary way
+        of getting the API to do the equivalent of clicking "Approve AIP
+        reingest" in the dashboard when faced with "Approve AIP reingest". This
+        is non-dry given ``approve_transfer_by_path_handler`` above.
+
+        [config]
+        name = approvePartialReingest
+        raise_exc = no
+        """
+        sip_uuid = payload.get("sip_uuid")
+        user_id = payload.get("user_id")
+        job = Job.objects.filter(
+            sipuuid=sip_uuid,
+            microservicegroup="Reingest AIP",
+            currentstep=Job.STATUS_AWAITING_DECISION,
+        ).first()
+        if not job:  # No job to be found.
+            raise NotFoundError(
+                'There is no "Reingest AIP" job awaiting a'
+                " decision for SIP {}".format(sip_uuid))
+        not_found = NotFoundError(
+            "Could not find choice for approve AIP reingest")
+        try:
+            chain = self.workflow.get_chain(self.APPROVE_AIP_REINGEST_CHAIN_ID)
+        except KeyError:
+            raise not_found
+        try:
+            choicesAvailableForUnits[job.pk].proceedWithChoice(
+                chain.id, user_id)
+        except IndexError:
+            raise not_found
+
+    def _get_processing_config_fields_handler(self, worker, job):
+        """List processing configuration fields.
+
+        [config]
+        name = getProcessingConfigFields
+        raise_exc = no
```

Comment:
This returns fairly nicely when `raise_exc` is set to `yes`.

Reply:
Yeah but the purpose of `raise_exc` is not having to change the original behaviour. Would you still do it?